src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1750,6 +1763,33 @@ int xcldev::xclReset(int argc, char *argv[])
         return -EINVAL;
     }
 
+    std::string vbnv, errmsg;
+    auto dev = pcidev::get_dev(index);
+    dev->sysfs_get( "rom", "VBNV", errmsg, vbnv );
+    if (!errmsg.empty()) {
+        std::cout << errmsg << std::endl;
+        return -EINVAL;
+    }
+    if (!all && vbnv.find("_u30_") != std::string::npos) {
```

Comment:
is _u30_ a stable interface?

Reply:
basically, this is just a workaround to provide card level reset just for u30.
and _u30_ alike keyword is used also in bandwidth.py to set BW target for different type of card

Reply:
I think that the idea here is that when U30 fixed this coupled hot reset issue, the platform team should allow us to tell the new u30 device from old ones so that new XRT can handle both old and new U30. The old XRT should not work on new U30 board/shell.

Reply:
Is there ANY other way to get this information instead of examining the name of the shell, if not then could you do the following:

1. Create CR regarding what is missing.  Note: I do believe that there is a confluence webpage that Anup created to track this type of "kludges" . 
2. Refactor the U30 code to its own method and call it from this method.

Reply:
already a CR for SC team